esmvaltool/diag_scripts/weighting/weighted_temperature_map.py

```diff
@@ -0,0 +1,188 @@
+
+
+def mapplot(dataarray, cfg, title_pattern, filename_part, ancestors, cmap=None, center=None):
+    """Visualize weighted temperature."""
+    period = '{start_year}-{end_year}'.format(**read_metadata(cfg)['tas'][0])
+    if 'tas_reference' in read_metadata(cfg).keys():
+        meta = read_metadata(cfg)['tas_reference']
+        period = 'change: {} minus {start_year}-{end_year}'.format(period, **meta[0])
+    metric = cfg['model_aggregation']
+    if isinstance(metric, int):
+        metric = f'{metric}perc'
+    proj = ccrs.PlateCarree(central_longitude=0)
+    fig, ax = plt.subplots(subplot_kw={'projection': proj})
+
+    dataarray.plot.pcolormesh(
+        ax=ax,
+        transform=ccrs.PlateCarree(),
+        levels=9,
+        robust=True,
+        extend='both',
+        cmap=cmap,
+        center=center,
+        # colorbar size often does not fit nicely
+        # https://stackoverflow.com/questions/18195758/set-matplotlib-colorbar-size-to-match-graph
+        # cbar_kwargs={'fraction': .021}
+    )
+
+    ax.coastlines()
+    lons = dataarray.lon.values
+    lats = dataarray.lat.values
+    longitude_formatter = LongitudeFormatter()
+    latitude_formatter = LatitudeFormatter()
+    if 'xticks' in cfg.keys():
+        ax.set_xticks(cfg['xticks'])
+    else:
+        ax.set_xticks(np.arange(np.floor(lons.min()), np.ceil(lons.max()), 10), crs=proj)
+    if 'yticks' in cfg.keys():
+        ax.set_yticks(cfg['yticks'])
+    else:
+        ax.set_yticks(np.arange(np.floor(lats.min()), np.ceil(lats.max()), 10), crs=proj)
+    ax.xaxis.set_ticks_position('both')
+    ax.yaxis.set_ticks_position('both')
+    ax.xaxis.set_major_formatter(longitude_formatter)
+    ax.yaxis.set_major_formatter(latitude_formatter)
+    ax.set_xlabel('')
+    ax.set_ylabel('')
+
+    title = title_pattern.format(metric=metric, period=period)
+    ax.set_title(title)
+
+    filename_plot = get_plot_filename(
+        filename_part, cfg)
+    fig.savefig(filename_plot, dpi=300, bbox_inches='tight')
+    plt.close(fig)
+
+    filename_data = get_diagnostic_filename(
+        filename_part, cfg, extension='nc')
+    dataarray.to_netcdf(filename_data)
+
+    log_provenance(title, filename_plot, cfg, ancestors)
+    log_provenance(title, filename_data, cfg, ancestors)
+
+
+def visualize_and_save_temperature(temperature: 'xr.DataArray',
+                                   cfg: dict,
+                                   ancestors: list):
+    """Wrapper for mapplot: absolute temperature"""
+    title_pattern = 'Weighted {metric} temperature \n{period} ($\degree$C)'
+    filename_part = 'temperature_change_weighted_map'
+    mapplot(temperature, cfg, title_pattern, filename_part, ancestors, cmap='Reds')
+
+
+def visualize_and_save_temperature_difference(temperature_difference: 'xr.DataArray',
+                                              cfg: dict,
+                                              ancestors: list):
+    """Wrapper for mapplot: temperature change"""
```

Comment:
I'm a bit confused by the terms temperature change and temperature difference. What you're using this for is actually the difference between weighted and unweighted, right? So it's basically the impact of weighting that you show. Do I get that right? Can we somehow make that clearer from the function name and/or docstring?